casper/src/main/scala/coop/rchain/casper/util/ProtoUtil.scala

```diff
@@ -410,23 +411,30 @@ object ProtoUtil {
     )
   }
 
-  def sourceDeploy(source: String, timestamp: Long): DeployData =
-    DeployData(user = ByteString.EMPTY, timestamp = timestamp, term = source)
+  def sourceDeploy(source: String, timestamp: Long, phlos: Int): DeployData =
+    DeployData(user = ByteString.EMPTY, timestamp = timestamp, term = source, phloLimit = phlos)
 
-  def compiledSourceDeploy(source: CompiledRholangSource, timestamp: Long): Deploy =
+  def compiledSourceDeploy(source: CompiledRholangSource, timestamp: Long, phloLimit: Int): Deploy =
     Deploy(
       term = Some(source.term),
-      raw = Some(sourceDeploy(source.code, timestamp))
+      raw = Some(sourceDeploy(source.code, timestamp, phloLimit))
     )
 
-  def termDeploy(term: Par, timestamp: Long): Deploy =
+  def termDeploy(term: Par, timestamp: Long, phloLimit: Int): Deploy =
```

Comment:
A good practice would be to use a wrapper type on top of Int, in order to take advantage of the type safety.

I'm thinking of something like this:

```scala
case class Cost(value:Integer)
```

This way we are sure we don't mix costs and timestamps.